### ISSUE TITLE
Error when trying to initialize SDK on Android.

### ISSUE THREAD

**Issue by danlfarnell** [vmwareairwatchsdk/vmware-ws1-sdk-forms#17]
Hi,

After much trial and error i'm finally able to get my Android app(iOS seems to work just fine) to compile when using Workspace One Forms nuget package.  However I can't seem to get the Sdk to initialize properly.  Any thoughts?  Thanks in advance.

When the following code is ran I get the following error listed below.

` WorkspaceOne.Android.WorkspaceOne.Instance.Init(this);`

[WorkspaceOne.Android.WorkspaceOne] AirWatchSDKException: Com.Airwatch.Sdk.AirWatchSDKException: AirWatch MDM Service Not Available
  at Java.Interop.JniEnvironment+StaticMethods.CallStaticObjectMethod (Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x00068] in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/Java.Interop/JniEnvironment.g.cs:12890 
  at Java.Interop.JniPeerMembers+JniStaticMethods.InvokeObjectMethod (System.String encodedMember, Java.Interop.JniArgumentValue* parameters) [0x00008] in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/Java.Interop/JniPeerMembers.JniStaticMethods.cs:151 
  at Com.Airwatch.Sdk.SDKManager.Init (Android.Content.Context context) [0x00031] in <4800cb4e9579403eb303ca160dd7b645>:0 
  at WorkspaceOne.Android.WorkspaceOne.Init (Android.App.Activity activity) [0x00028] in <0aba87adb8fb45c1b5939cb3730d0a47>:0 
  --- End of managed Com.Airwatch.Sdk.AirWatchSDKException stack trace ---
com.airwatch.sdk.AirWatchSDKException: AirWatch MDM Service Not Available
	at com.airwatch.sdk.SDKManager.getInstance(SDKManager.java:572)
	at com.airwatch.sdk.SDKManager.init(SDKManager.java:298)
	at crc64e8e4ebd59417f5fb.MainActivity.n_onCreate(Native Method)
	at crc64e8e4ebd59417f5fb.MainActivity.onCreate(MainActivity.java:53)
	at android.app.Activity.performCreate(Activity.java:8326)
	at android.app.Activity.performCreate(Activity.java:8306)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1347)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4212)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4403)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:104)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2498)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8855)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)

When I call the following line right after that, an exception is thrown.

`WorkspaceOne.Android.WorkspaceOne.Instance.OnCreate(bundle);`

e.Message = "KoinApplication has not been started"

**Comment by Maddy79** [comment]
@danlfarnell  / @Pelinalpp  - we will check and update.

**Comment by akalghatgi** [comment]
@danlfarnell / @Pelinalpp Can you please let me the mode of Deployment of APK on Device, 
1. Running in Debug mode directly on Device? OR
2. Installing App from Hub/Catalog?

Regards,
Amit

**Comment by Pelinalpp** [comment]
I got release apk and uploaded it to the Hub/Catalog. But it throws "Java.Lang.IllegalStateException: KoinApplication has not been started" error.

**Comment by akalghatgi** [comment]
@Pelinalpp Thanks... Can you please provide the complete logs also VS version and Library Reference details. Or if you can provide us with Sample App code... replicating the issue... 

Thanks & Regards,
Amit Kalghatgi

**Comment by Pelinalpp** [comment]
I am sorry I think I fixed koin error. I forgot to add [Application] to MainApplication class. I pushed to github repo of my sample project. I am so sorry.
However my issue is continue. Now error is changed but there is new error. What could be cause of this error?

Error:
com.airwatch.crypto.openssl.OpenSSLWrapper.<init>
java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "file_encrypt_rsa" referenced by "/data/app/com.companyname.xamarinawsdk-5gp2nZirCv0ZFpyXUHIzLw==/lib/arm64/libfips_main.so"...

java.lang.Runtime.loadLibrary0 Runtime.java:1016
java.lang.System.loadLibrary System.java:1657
com.airwatch.crypto.openssl.OpenSSLWrapper.<init>
com.airwatch.crypto.openssl.OpenSSLCryptUtil.<init>
com.airwatch.crypto.openssl.OpenSSLCryptUtil.createInstance
com.airwatch.keymanagement.unifiedpin.token.TokenFileStorageImpl.<init>
com.airwatch.keymanagement.unifiedpin.token.DefaultTokenFactory.<init>
com.airwatch.keymanagement.unifiedpin.DefaultUnifiedPinContext$mTokenFactory$2.invoke
com.airwatch.keymanagement.unifiedpin.DefaultUnifiedPinContext$mTokenFactory$2.invoke
kotlin.SynchronizedLazyImpl.getValue LazyJVM.kt:74
com.airwatch.keymanagement.unifiedpin.DefaultUnifiedPinContext.getMTokenFactory
com.airwatch.keymanagement.unifiedpin.DefaultUnifiedPinContext.getTokenFactory
com.airwatch.app.AWSDKApplicationDelegate.getTokenFactory
com.airwatch.app.AWApplication.getTokenFactory
com.airwatch.keymanagement.unifiedpin.UnifiedPinHandler.initSDKContext
com.airwatch.keymanagement.unifiedpin.UnifiedPinHandler.handle
com.airwatch.sdk.context.awsdkcontext.handlers.SDKBaseHandler.handleNextHandler
com.airwatch.sdk.context.awsdkcontext.handlers.DeviceIDPermissionUiHandler.handle
com.airwatch.sdk.context.awsdkcontext.handlers.SDKBaseHandler.handleNextHandler
com.airwatch.sdk.context.awsdkcontext.handlers.EMMValidationHandler.handle
com.airwatch.sdk.context.awsdkcontext.handlers.SDKBaseHandler.handleNextHandler
com.airwatch.sdk.context.awsdkcontext.handlers.standalone.FetchSrvDetailsP2PHandler.handle
com.airwatch.sdk.context.awsdkcontext.handlers.SDKBaseHandler.handleNextHandler
com.airwatch.sdk.context.awsdkcontext.handlers.anchor.FetchSrvDetailsAnchor.resolveSSOMode
com.airwatch.sdk.context.awsdkcontext.handlers.anchor.FetchSrvDetailsAnchor.onSuccess
com.airwatch.sdk.context.awsdkcontext.SDKContextCallable.onSuccess
com.airwatch.sdk.context.awsdkcontext.SDKContextCallable.onSuccess
com.airwatch.task.CallbackFuture.lambda$onSuccess$0
com.airwatch.task.CallbackFuture$$ExternalSyntheticLambda1.run
android.os.Handler.handleCallback Handler.java:789
android.os.Handler.dispatchMessage Handler.java:98
android.os.Looper.loop Looper.java:164
android.app.ActivityThread.main ActivityThread.java:6944
java.lang.reflect.Method.invoke Method.java
com.android.internal.os.Zygote$MethodAndArgsCaller.run Zygote.java:327
com.android.internal.os.ZygoteInit.main ZygoteInit.java:1374

com.airwatch.crypto.openssl.OpenSSLCryptUtil.<init>
com.airwatch.crypto.openssl.OpenSSLCryptUtil.createInstance
com.airwatch.keymanagement.unifiedpin.token.TokenFileStorageImpl.<init>
com.airwatch.keymanagement.unifiedpin.token.DefaultTokenFactory.<init>
com.airwatch.keymanagement.unifiedpin.DefaultUnifiedPinContext$mTokenFactory$2.invoke
com.airwatch.keymanagement.unifiedpin.DefaultUnifiedPinContext$mTokenFactory$2.invoke
kotlin.SynchronizedLazyImpl.getValue LazyJVM.kt:74
com.airwatch.keymanagement.unifiedpin.DefaultUnifiedPinContext.getMTokenFactory
com.airwatch.keymanagement.unifiedpin.DefaultUnifiedPinContext.getTokenFactory
com.airwatch.app.AWSDKApplicationDelegate.getTokenFactory
com.airwatch.app.AWApplication.getTokenFactory
com.airwatch.keymanagement.unifiedpin.UnifiedPinHandler.initSDKContext
com.airwatch.keymanagement.unifiedpin.UnifiedPinHandler.handle
com.airwatch.sdk.context.awsdkcontext.handlers.SDKBaseHandler.handleNextHandler
com.airwatch.sdk.context.awsdkcontext.handlers.DeviceIDPermissionUiHandler.handle
com.airwatch.sdk.context.awsdkcontext.handlers.SDKBaseHandler.handleNextHandler
com.airwatch.sdk.context.awsdkcontext.handlers.EMMValidationHandler.handle
com.airwatch.sdk.context.awsdkcontext.handlers.SDKBaseHandler.handleNextHandler
com.airwatch.sdk.context.awsdkcontext.handlers.standalone.FetchSrvDetailsP2PHandler.handle
com.airwatch.sdk.context.awsdkcontext.handlers.SDKBaseHandler.handleNextHandler
com.airwatch.sdk.context.awsdkcontext.handlers.anchor.FetchSrvDetailsAnchor.resolveSSOMode
com.airwatch.sdk.context.awsdkcontext.handlers.anchor.FetchSrvDetailsAnchor.onSuccess
com.airwatch.sdk.context.awsdkcontext.SDKContextCallable.onSuccess
com.airwatch.sdk.context.awsdkcontext.SDKContextCallable.onSuccess
com.airwatch.task.CallbackFuture.lambda$onSuccess$0
com.airwatch.task.CallbackFuture$$ExternalSyntheticLambda1.run
android.os.Handler.handleCallback Handler.java:789
android.os.Handler.dispatchMessage Handler.java:98
android.os.Looper.loop Looper.java:164
android.app.ActivityThread.main ActivityThread.java:6944
java.lang.reflect.Method.invoke Method.java
com.android.internal.os.Zygote$MethodAndArgsCaller.run Zygote.java:327
com.android.internal.os.ZygoteInit.main ZygoteInit.java:1374


com.airwatch.keymanagement.unifiedpin.token.TokenFileStorageImpl.<init>
com.airwatch.keymanagement.unifiedpin.token.DefaultTokenFactory.<init>
com.airwatch.keymanagement.unifiedpin.DefaultUnifiedPinContext$mTokenFactory$2.invoke
com.airwatch.keymanagement.unifiedpin.DefaultUnifiedPinContext$mTokenFactory$2.invoke
kotlin.SynchronizedLazyImpl.getValue LazyJVM.kt:74
com.airwatch.keymanagement.unifiedpin.DefaultUnifiedPinContext.getMTokenFactory
com.airwatch.keymanagement.unifiedpin.DefaultUnifiedPinContext.getTokenFactory
com.airwatch.app.AWSDKApplicationDelegate.getTokenFactory
com.airwatch.app.AWApplication.getTokenFactory
com.airwatch.keymanagement.unifiedpin.UnifiedPinHandler.initSDKContext
com.airwatch.keymanagement.unifiedpin.UnifiedPinHandler.handle
com.airwatch.sdk.context.awsdkcontext.handlers.SDKBaseHandler.handleNextHandler
com.airwatch.sdk.context.awsdkcontext.handlers.DeviceIDPermissionUiHandler.handle
com.airwatch.sdk.context.awsdkcontext.handlers.SDKBaseHandler.handleNextHandler
com.airwatch.sdk.context.awsdkcontext.handlers.EMMValidationHandler.handle
com.airwatch.sdk.context.awsdkcontext.handlers.SDKBaseHandler.handleNextHandler
com.airwatch.sdk.context.awsdkcontext.handlers.standalone.FetchSrvDetailsP2PHandler.handle
com.airwatch.sdk.context.awsdkcontext.handlers.SDKBaseHandler.handleNextHandler
com.airwatch.sdk.context.awsdkcontext.handlers.anchor.FetchSrvDetailsAnchor.resolveSSOMode
com.airwatch.sdk.context.awsdkcontext.handlers.anchor.FetchSrvDetailsAnchor.onSuccess
com.airwatch.sdk.context.awsdkcontext.SDKContextCallable.onSuccess
com.airwatch.sdk.context.awsdkcontext.SDKContextCallable.onSuccess
com.airwatch.task.CallbackFuture.lambda$onSuccess$0
com.airwatch.task.CallbackFuture$$ExternalSyntheticLambda1.run
android.os.Handler.handleCallback Handler.java:789
android.os.Handler.dispatchMessage Handler.java:98
android.os.Looper.loop Looper.java:164
android.app.ActivityThread.main ActivityThread.java:6944
java.lang.reflect.Method.invoke Method.java
com.android.internal.os.Zygote$MethodAndArgsCaller.run Zygote.java:327
com.android.internal.os.ZygoteInit.main ZygoteInit.java:1374

**Comment by akalghatgi** [comment]
@Pelinalpp are you trying to run in debug mode on device directly? create a app and upload to catalog and try..

Thanks & Regards,
Amit

**Comment by Pelinalpp** [comment]
Hi Amit,
Actually I tried with upload apk file to the catalog. After a few tries, this error has disappeared.

When I get the apk of the my sample project (the project in my github account: https://github.com/Pelinalpp/XamarinAWSDK) and distribute it to the devices on the catalog, I can open the application, but I see that I can't get any dlp rules. However, I added sdk and application profiles while deploying the application in VMWare.

Also, when I distributed this apk from vmware, I saw that 2 icons appeared on the phone. (1 is Android AW Network Library and the other is the icon of my sample project. I did not understand why this happened.
![Screenshot_20221023-183730_Gallery](https://user-images.githubusercontent.com/24781631/197402969-13827342-a460-4461-abc4-05999cffd3b1.jpg)

But when I tried to open the Android AW Network Library application, the application did not open as seen in the video. When I tried to open my own sample project, the application opened but I cannot get the sdk profiles)
In addition, I added the crash log of the app center to my application to see if there is an error and I saw that the following error occurred.
```
SDKBaseActivityDelegate.OnResume ()
Java.Lang.RuntimeException: Unable to find explicit activity class {com.companyname.xamarinawsdk/com.airwatch.login.ui.activity.SDKSplashActivity}; have you declared this activity in your AndroidManifest.xml?
```

I downloaded the this issue’s sample project and changed only the package name and version/build number (I set the package name the same as the package name of the sample project I created). In this way, I got the apk and distributed the application from the Workspace One UEM panel. When I try to open the application on the Android device, I get the error that occurs in the images below. (I also started seeing only the Android AW Network Library icon on the phone.)
![Screenshot_20221023-183606_Gallery](https://user-images.githubusercontent.com/24781631/197402987-2c6ec208-d490-4943-b1d6-2b928a455452.jpg)
![Screenshot_20221023-183708_Gallery](https://user-images.githubusercontent.com/24781631/197402997-da85f3be-5f98-4c85-94e1-91bdb7fe3ce3.jpg)

In order to see the error, I also set up appcenter crash this app. I put logs especially where I think there may be problems. However, there are no logs.

In addition I added the video what happens with drive link.
https://drive.google.com/drive/folders/1TF7q1sot-bTr2sR2-79iY2BCwZ-fDHbS?usp=sharing
I'm sorry, I know it's been too long, but I couldn't find how to explain it more simply.

Thanks.
Pelin Konaray.

**Comment by Pelinalpp** [comment]
Hi, 

Problem was fixed with added the following line to the app.xaml.cs
`ws.FormsDelegate = this;`
![image (4)](https://user-images.githubusercontent.com/24781631/199432249-01f4ab16-436a-428f-8f17-92a767c16ce2.png)

This causes android error. Maybe this repo must be fixed. 
Because there are scattered and few resources for Xamarin.Forms VMWare SDK intagration. And it's so hard to find solution.
Maybe this help to someone.

Thanks,
Regards.

**Comment by danlfarnell** [comment]
So that appeared to be my issue as well.  I didn't have the [Application] tag on the Application class.  Thanks to @Pelinalpp for pointing that out :).  I have encountered another error after that however, but I will open another issue for this.  Thanks for your assistance.
